### PR TITLE
Add queries to language service

### PR DESCRIPTION
**Pull request by michaelmarshall**
Added full CRUD to language service, including `update` with optional params. I might have unnecessarily modified too many interfaces or DTOs with nullables, so I'm sure there are some changes that could be removed.

### CODE REVIEW

File: apps/api/src/app/languages/dto/create-language.dto.ts [hunk 1]
```diff
@@ -1,12 +1,26 @@
...
+  @Field(() => Int)
+  beginFiscalYear: number;
+  @Field()
+  ethnologueName: string;
+  @Field(() => Int)
+  ethnologuePopulation: number;
+  @Field(() => Int)
+  organizationPopulation: number;
+  @Field(() => Int)
+  rodNumber?: number;
```
Comment (CarsonF) [inline apps/api/src/app/languages/dto/create-language.dto.ts]:
```suggestion
  rodNumber: number;
```

File: apps/api/src/app/languages/language.service.ts [hunk 2]
```diff
@@ -1,14 +1,27 @@
...
+      })
+      .return([
+        'lang.id as id',
+        'lang.name as name',
+        'lang.displayName as displayName',
+        'lang.beginFiscalYear as beginFiscalYear',
+        'lang.ethnologueName as ethnologueName',
+        'lang.ethnologuePopulation as ethnologuePopulation',
+        'lang.organizationPopulation as organizationPopulation',
+        'lang.rodNumber as rodNumber',
```
Comment (CarsonF) [inline apps/api/src/app/languages/language.service.ts]: I wanted to bring this up yesterday but didn't. Wouldn't it just be easier to return the `lang` node?

File: apps/api/src/app/languages/language.ts [hunk 3]
```diff
@@ -6,26 +6,26 @@ export class Language implements ILanguage {
...
 
-  @Field(() => Int)
-  organizationPopulation: number;
+  @Field(() => Int, { nullable: true })
+  organizationPopulation: number | null;
 
-  @Field(() => Int)
-  rodNumber: number;
+  @Field(() => Int, { nullable: true })
+  rodNumber: number | null;
```
Comment (CarsonF) [inline apps/api/src/app/languages/language.ts]: If we are requiring all these fields on create then they don't need to be nullable/

File: apps/api/src/app/languages/language.service.ts [hunk 4]
```diff
@@ -28,7 +64,105 @@ export class LanguageService {
...
+      .raw('SET lang.active = false');
     await query.run();
   }
+
+  async update(input: UpdateLanguage): Promise<Language> {
+    /*
+    it is preferable to run the query in one go, but some
+    fields may be null. We need to check each field and 
+    create the correct query.
+    */
```
Comment (CarsonF) [inline apps/api/src/app/languages/language.service.ts]: Yes I see what you mean. Let me take a few mins come up with a better solution. This will not scale out.

File: apps/api/src/app/languages/language.ts (hunk 3 above)
Comment (michaelmarshall) [reply apps/api/src/app/languages/language.ts]: right.

File: apps/api/src/app/languages/language.service.ts (hunk 2 above)
Comment (michaelmarshall) [reply apps/api/src/app/languages/language.service.ts]: yes, but it seems graphql wants the return identifiers named by their schema property names instead of the native way neo4j returns a node's properties. Native return would be `lang.name`, but graphql wants `name`

Comment (michaelmarshall) [reply apps/api/src/app/languages/language.service.ts]: I'm not aware of a shortcut in cypher to get there without explicit mappings in the return statement.

Comment (CarsonF) [reply apps/api/src/app/languages/language.service.ts]: I had a commit that did it but didn't bother showing you since we had already pressed hard on the query builder.

But basically it was
```ts
const node: Node<Language> = await query.return('lang').first();
return Language.from(node.properties);
```
`node` has `properties`, `labels` and `identity`.

maybe it's worth pursuing further, but I figured most of our queries wouldn't have a single node with all the properties we need.

Seems like it's the same return for most queries though so maybe that part could be abstracted so it's only done once per high level object.

File: apps/api/src/app/languages/language.ts (hunk 3 above)
Comment (CarsonF) [reply apps/api/src/app/languages/language.ts]: Ok so we can remove that change.

File: apps/api/src/app/languages/language.service.ts [hunk 5]
```diff
@@ -19,7 +32,30 @@ export class LanguageService {
...
+          id: input.id,
+          name: input.name,
+          displayName: input.displayName,
+          beginFiscalYear: input.beginFiscalYear,
+          ethnologueName: input.ethnologueName,
+          ethnologuePopulation: input.ethnologuePopulation,
+          organizationPopulation: input.organizationPopulation,
+          rodNumber: input.rodNumber,
+        }
+      )
```
Comment (CarsonF) [inline apps/api/src/app/languages/language.service.ts]: FYI this is what it could look like:
```ts
.merge(node('lang', 'Language', { ...input, active: true }))
```

File: apps/api/src/app/languages/language.service.ts [hunk 6]
```diff
@@ -70,85 +71,14 @@ export class LanguageService {
...
-    }
-
+  async update({ id, ...input }: UpdateLanguage): Promise<Language> {
     const query = this.neo4j
       .raw('MATCH (lang:Language {id: $id, active: true})', {
-        id: input.id,
+        id,
+      })
+      .setValues({
+        lang: pickBy(input),
```
Comment (CarsonF) [inline apps/api/src/app/languages/language.service.ts]: `pickBy` omits properties with falsy values.
```ts
.setValues({ lang: {...} })
```
will generate the `set lang.prop = $prop, ...` for us.

File: apps/api/src/app/languages/language.service.ts [hunk 7]
```diff
@@ -28,7 +65,34 @@ export class LanguageService {
...
   }
+
+  async update({ id, ...input }: UpdateLanguage): Promise<Language> {
+    const query = this.neo4j
+      .raw('MATCH (lang:Language {id: $id, active: true})', {
+        id,
+      })
+      .setValues({
+        lang: pickBy(input),
+      })
```
Comment (CarsonF) [inline apps/api/src/app/languages/language.service.ts]: `pickBy` omits properties with falsy values.

```ts
.setValues({ lang: {...} })
```
will generate the `set lang.prop = $prop, ...` for us.

File: apps/api/src/app/languages/language.service.ts (hunk 4 above)
Comment (michaelmarshall) [reply apps/api/src/app/languages/language.service.ts]: @CarsonF any new thoughts?

File: apps/api/src/app/languages/language.service.ts (hunk 7 above)
Comment (michaelmarshall) [reply apps/api/src/app/languages/language.service.ts]: Did you test this? It doesn't work as far as I can tell.

Comment (CarsonF) [reply apps/api/src/app/languages/language.service.ts]: Let me try again

Comment (CarsonF) [reply apps/api/src/app/languages/language.service.ts]: Ok it was not spreading out the properties
```
SET lang = $lang
```
So anything not included was null'ed.

I changed it to merge and it's all good now
```
SET lang += $lang
```
